bgpd/bgp_mplsvpn.c

```diff
@@ -1598,6 +1598,16 @@ vpn_leak_from_vrf_get_per_nexthop_label(afi_t afi, struct bgp_path_info *pi,
 		return bgp_mplsvpn_get_vpn_label(&from_bgp->vpn_policy[afi]);
 	}
 
+	if (is_bgp_static_route &&
+	    pi->nexthop->nexthop->type == NEXTHOP_TYPE_IFINDEX) {
```

Comment:
why not check against pi->sub_type being BGP_TYPE_REDISTRIBUTED? Also, aggregated routes should be taken into account?

Reply:
I want to target only the manually added prefixes using the "network" command in BGP (`bgp_static_set`). I am not considering routes that were redistributed from other protocols or created through aggregation.

Reply:
I meant BGP_TYPE_STATIC, not redistributed :)

Reply:
```
 is_bgp_static_route = bpi_ultimate->sub_type == BGP_ROUTE_STATIC &&
 ›       ›             bpi_ultimate->type == ZEBRA_ROUTE_BGP;       
```

The check is made in `is_bgp_static_route`

Reply:
Then what is the reason of this check `pi->nexthop->nexthop->type == NEXTHOP_TYPE_IFINDEX`?

Reply:
By adding a connected prefix using the "network" command in a VRF, a nexthop cache `bnc` is created with` bnc->prefix ` set to the prefix itself. This entry is then registered with Zebra to resolve the nexthop. This results in adding a nexthop via the VRF loopback interface (thus of type NEXTHOP_TYPE_IFINDEX) to the `bnc->nexhop` list.

Compared to other types of routes, which use `bnc->prefix` learned from BGP updates, connected routes inside a VRF do not have BGP updates. Therefore, we need to route them to the VRF loopback interface.